components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -670,7 +670,7 @@ abstract class AbstractFetchDownloadService : Service() {
         val downloadWithUniqueFileName = makeUniqueFileNameIfNecessary(download, append)
         updateDownloadState(downloadWithUniqueFileName)
 
-        if (SDK_INT >= Build.VERSION_CODES.Q) {
+        if (SDK_INT >= Build.VERSION_CODES.Q && !Environment.isExternalStorageLegacy()) {
```

Comment:
Let's add tests to verify this behaviour :)

Reply:
@Amejia481 I would, but there are 2 issues:
- We don't use PowerMock, so besides giving the whole `Environment.isExternalStorageLegacy()` as a boolean param, which I don't feel like we should do, I don't know exactly how to mock that
- We need to update Roboelectric to version 4.3.1 to support Android Q which comes with some issues, and from what I see here: https://github.com/robolectric/robolectric/issues/5245 there's a workaround available, but excluding a library in all our AC modules seems not worth it

So I think we'll have to skip testing it for now

Reply:
That's correct. There is an alternative that we could use, it's to wrap `isExternalStorageLegacy` in a internal function on `AbstractFetchDownloadService` and then `spy` on `AbstractFetchDownloadService` and change the return of the function, for example:

```kotlin
abstract class AbstractFetchDownloadService : Service() {
   @VisibleForTesting
    internal fun isExternalStorageLegacy(): Boolen{
        return Environment.isExternalStorageLegacy()
   }
}


```


```kotlin

    @Test
    fun `example`() = runBlocking {

        val mockStore = mock<BrowserStore>()
        val service = spy(object : AbstractFetchDownloadService() {
            override val httpClient = client
            override val store = mockStore
        })

        doReturn(false).`when`(service).isExternalStorageLegacy()
       
         assertFalse(service.isExternalStorageLegacy())
       
        doReturn(true).`when`(service).isExternalStorageLegacy()

        assertTrue(service.isExternalStorageLegacy())
    }
```

Reply:
@Amejia481 you're right, I added tests, thanks!

I'm usually not a fan of adding wrapper methods just for testing, but here it makes sense, since it balances the abstraction inside the method.